### PR TITLE
Kraken: Error on journey request with arrival before due to double use of speed_factor and trimmed float value of duration

### PR DESCRIPTION
**Pull request by kadhikari**

### CODE REVIEW

**Comment by kadhikari** [comment]
Concerned ticket: http://jira.canaltp.fr/browse/ITI-287

File: source/jormungandr/jormungandr/interfaces/v1/fields.py [hunk 1]
```diff
@@ -465,6 +466,27 @@ def output(self, key, region):
...
+class Integer(fields.Raw):
+    """
+    Integer with round value
+    2.3 -> 2
+    2.6 -> 3
+    """
+    def __init__(self, default=0, **kwargs):
+        super(Integer, self).__init__(default=default, **kwargs)
+
+    def output(self, key, obj):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/fields.py]: hum I think it's better to overload the format method for this no ?

File: source/jormungandr/tests/check_utils.py [hunk 2]
```diff
@@ -625,7 +625,8 @@ def is_valid_section(section, query):
             assert dur >= 0
             total_duration += dur
 
-        assert total_duration == section['duration']
+        #assert total_duration == section['duration']
+        assert total_duration - section['duration'] <= 2
```
Comment (antoine-de) [inline source/jormungandr/tests/check_utils.py]: you should add a comment there

and I don't understand the `2`. isn't it dependent on the `len(section['path']` ?

Comment (TeXitoi) [reply source/jormungandr/tests/check_utils.py]:
```python
assert abs(total_duration - section['duration']) <= 0.5 * len(section['path'])
```

File: source/routing/tests/routing_api_test_data.h [hunk 3]
```diff
@@ -496,6 +496,10 @@ struct routing_api_data {
         return speed_trait.to_duration(dist, mode);
     }
 
+    double to_precise_duration(float dist, navitia::type::Mode_e mode) {
```
Comment (TeXitoi) [inline source/routing/tests/routing_api_test_data.h]: I'd add a `double navitia::time_duration::total_fractional_seconds() const` to `navitia::time_duration`. That way you can easily use that in tests and code.

File: source/jormungandr/tests/journey_common_tests.py [hunk 4]
```diff
@@ -119,14 +119,14 @@ def test_speed_factor_direct_path(self):
         non_pt_walk_j = next((j for j in journeys if j['type'] == 'non_pt_walk'), None)
         assert non_pt_walk_j
         assert non_pt_walk_j['duration'] == non_pt_walk_j['sections'][0]['duration']
-        assert non_pt_walk_j['duration'] == 205
-        assert non_pt_walk_j['durations']['total'] == 205
-        assert non_pt_walk_j['durations']['walking'] == 205
+        assert non_pt_walk_j['duration'] == 206
```
Comment (TeXitoi) [inline source/jormungandr/tests/journey_common_tests.py:123]: why does that change ?

File: source/jormungandr/tests/check_utils.py (hunk 2 above)
Comment (kadhikari) [reply source/jormungandr/tests/check_utils.py]: i modified as : assert abs(total_duration - section['duration']) <= 0.5 * len(section['path']) + 1

Comment (kadhikari) [reply source/jormungandr/tests/check_utils.py]: i hope it is also clear for Antoine

**Comment by kadhikari** [comment]
@antoine-de @TeXitoi I push with some corrections

**Review by TeXitoi — APPROVED** [review]

**Comment by kadhikari** [comment]
jenkins retest this please

**Comment by TeXitoi** [comment]
Jenkins, test this please

**Comment by kadhikari** [comment]
jenkins, test this please

**Comment by kadhikari** [comment]
jenkins, test this please

**Pull request merged**